haystack/schema.py

```diff
@@ -533,17 +533,18 @@ def __init__(
         # If an Answer is provided we need to make sure that it's consistent with the `no_answer` value
         # TODO: reassess if we want to enforce Span.start=0 and Span.end=0 for no_answer=True
         if self.answer is not None:
+            if no_answer is None:
+                # Set default before validation
+                no_answer = self.answer.answer == "" or self.answer.answer is None
```

Comment:
This change is definitely correct, but I have mixed feelings about it :smile: 

Adding this check here seems to void entirely the purpose of having a separate `no_answer` field in `Label`. Which might be perfectly fine: we just need to take a small decision and be consistent with it.

I tend to believe that this `no_answer` field is not very useful (I mean, just do `if self.answer` to see if there's an answer, right?), so we might go for a quick attempt at removing it.  Let's try to do this first.

However, this might become a large refactoring and become messy quite fast. If that happens, we can still help the situation by transforming `no_answer` into a read-only `@property` and get rid of this whole validation issue. `no_answer` will simply reflect the content of `answer` and look like:

```python
@property
def no_answer(self) -> bool:
    return self.answer is None or self.answer.strip() == ""
```
(note that if we add only this code, `no_answer` is automatically read-only).

Let me know if you are up for this :blush:

Reply:
@wochinge any additional thoughts? Would you be ok with this refactoring or you see any practical use for a separate, independent `no_answer` field?

Reply:
@ZanSara I have limited myself to reporting in this PR what @wochinge proposed, but in principle, I agree that it does not make much sense to validate an automatically populated field...

I await your opinion before going on...

Reply:
That was a good approach, no worries. I just like a lot to take this occasion to simplify and improve the code where we can :grin:

I think you can go ahead with this change!

Reply:
Sorry I didn't understand what you suggest in conclusion...

Reply:
Sorry for the late reply. Let's try to remove `no_answer` :slightly_smiling_face:

Reply:
I'm a coward and don't want to spend the next few months on this PR. :smiley: 
Seeing how much `no_answer` is used in the codebase, I try the property solution. 
Anyway, I'm afraid it won't be a walk in the park...

Reply:
Hey no pressure! The property is already a fantastic step forward! :rocket: